### PR TITLE
Plugin 1.3.4

### PR DESCRIPTION
**Pull request by Mygod**
Main change: support plugin aliases proposed by @CzBiX: https://github.com/shadowsocks/v2ray-plugin-android/issues/18#issuecomment-581262267
For other changes, see CHANGES.md.

I have not thoroughly tested this. @madeye Can you help? (client app should not require a plugin lib update)

### CODE REVIEW

**Comment by madeye** [comment]
To test this change, I think we need to update the plugin app accordingly?

**Comment by Mygod** [comment]
Done.

**Comment by Mygod** [comment]
To test 4a6dc8049c71f537059215685700ba5e33baf633, we could test against [GoQuiet-android](https://github.com/cbeuw/GoQuiet-android).

**Comment by madeye** [comment]
I tested it locally and found one issue.

In the profile fragment, after switching the plugin, the option didn't get updated. Then saving the profile and open the profile again, the plugin option is updated.

File: mobile/src/main/java/com/github/shadowsocks/ProfileConfigFragment.kt [hunk 1]
```diff
@@ -102,18 +103,6 @@ class ProfileConfigFragment : PreferenceFragmentCompat(),
         findPreference<Preference>(Key.udpdns)!!.isEnabled = serviceMode != Key.modeProxy
         plugin = findPreference(Key.plugin)!!
         pluginConfigure = findPreference(Key.pluginConfigure)!!
-        plugin.unknownValueSummary = getString(R.string.plugin_unknown)
-        plugin.setOnPreferenceChangeListener { _, newValue ->
```
Comment (madeye) [inline mobile/src/main/java/com/github/shadowsocks/ProfileConfigFragment.kt]: Why removed `setOnPreferenceChangeListener` here?

Comment (Mygod) [reply mobile/src/main/java/com/github/shadowsocks/ProfileConfigFragment.kt]: Hmm it is supposed to be handled by `onActivityResult` now.

Comment (madeye) [reply mobile/src/main/java/com/github/shadowsocks/ProfileConfigFragment.kt]: Do you where the plugin icon and name updated when the plugin is switch in the dialog? I'm debugging the plugin switching issue.

File: mobile/src/main/java/com/github/shadowsocks/ProfileConfigFragment.kt [hunk 2]
```diff
@@ -221,6 +206,19 @@ class ProfileConfigFragment : PreferenceFragmentCompat(),
...
+                val override = pluginConfiguration.pluginsOptions.keys.firstOrNull {
+                    plugin.plugins.lookup[it] == selected
+                }
+                pluginConfiguration = PluginConfiguration(pluginConfiguration.pluginsOptions, override ?: selected.id)
+                DataStore.plugin = pluginConfiguration.toString()
+                DataStore.dirty = true
+                pluginConfigure.isEnabled = selected !is NoPlugin
+                pluginConfigure.text = pluginConfiguration.getOptions().toString()
+                if (!selected.trusted) Snackbar.make(view!!, R.string.plugin_untrusted, Snackbar.LENGTH_LONG).show()
+            }
```
Comment (Mygod) [inline mobile/src/main/java/com/github/shadowsocks/ProfileConfigFragment.kt]: @madeye This is supposed to be the callback.

File: mobile/src/main/java/com/github/shadowsocks/ProfileConfigFragment.kt (hunk 1 above)
Comment (Mygod) [reply mobile/src/main/java/com/github/shadowsocks/ProfileConfigFragment.kt]: Committed a fix via web editor. Let me know if it works. :)

**Comment by Mygod** [comment]
Cool. Let's merge this for the next release.

**Pull request merged**